### PR TITLE
replayParser refactor

### PR DESCRIPTION
**Pull request by theoculus8**
I found some of the code in replayParser.cs a little hard to follow, so I refactored it to make it easier to follow. The code is less efficient than Yucheng's code, but uses a little less memory.

### CODE REVIEW

**Review by yatzyt — COMMENTED** [review]
If it doesn't break anything or affect performance, I don't see why not.

**Review by Haochenlucas — APPROVED** [review]
Look ok to me.

**Review by dominate0704 — APPROVED** [review]
it looks fine
